Snippets/Snippets_5/Features/FeatureConfiguration.cs

```diff
@@ -0,0 +1,72 @@
+﻿namespace Snippets5.Features
+{
+    using System;
+    using NServiceBus;
+    using NServiceBus.Features;
+    using NServiceBus.Pipeline;
+
+    class MyMagicFeature : Feature
+    {
+
+#region FeatureConfiguration
+        public MyMagicFeature()
+        {
+            EnableByDefault();
+
+            DependsOn<Sagas>();
+
+            Prerequisite(c => c.Settings.EndpointName().StartsWith("Magical"), "This feature only works on magical endpoints");
+        }
+#endregion
+
+#region FeatureSetup
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            // retrieve some global settings
+            var supervised = context.Settings.Get<bool>("SupervisorActive");
+
+            if (!supervised)
```

Comment:
minor coding style suggestion: invert this. if a negated and a non negated if result in roughly equivalent code u should always chose the non negated since it is easier for the developer reading to not need to do the extra mental operation.